### PR TITLE
Add environment secrets to docs

### PR DESCRIPTION
**Pull request by letsbelopez**
Add docs for environment secrets via parameter store.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: doc_source/environment-variables.md [hunk 1]
```diff
@@ -115,4 +115,37 @@ You can use the following environment variables that are accessible by default w
...
+
+**To set environment secrets**
+
+1. Sign in to the AWS Management Console and open the [AWS Systems Manager Console](https://console.aws.amazon.com/systems-manager/)\.
+
+1. In the AWS Systems Manager Console, choose Parameter Store, and then choose Create Parameter.
+
+1. Create a parameter name in the format `/amplify/{your_app_id}/{your_backend_environment_name}/{your_parameter_name}`
+
+1. Under type, choose Secure String for encrypted values
```
Comment (pananapread) [inline doc_source/environment-variables.md]: One word and bold "SecureString"

File: doc_source/environment-variables.md [hunk 2]
```diff
@@ -115,4 +115,37 @@ You can use the following environment variables that are accessible by default w
...
+
+1. Under KMS key source, choose my currenct account to use the default key for your account
```
Comment (pananapread) [inline doc_source/environment-variables.md]: "My current account" check spelling + bold

File: doc_source/environment-variables.md [hunk 3]
```diff
@@ -115,4 +115,37 @@ You can use the following environment variables that are accessible by default w
...
+
+1. Under Value, enter the value you want
+
+1. Click Create parameter
```
Comment (pananapread) [inline doc_source/environment-variables.md]: bold **Create parameter**

File: doc_source/environment-variables.md [hunk 4]
```diff
@@ -115,4 +115,37 @@ You can use the following environment variables that are accessible by default w
...
+
+Environment secrets are similar to environment variables, but they are SSM parameter store key value pairs and can be encrypted.
+
+## Set environment secrets<a name="setting-env-secrets"></a>
+
+**To set environment secrets**
+
+1. Sign in to the AWS Management Console and open the [AWS Systems Manager Console](https://console.aws.amazon.com/systems-manager/)\.
+
+1. In the AWS Systems Manager Console, choose Parameter Store, and then choose Create Parameter.
```
Comment (pananapread) [inline doc_source/environment-variables.md]: Bold "Parameter Store" and "Create parameter"

File: doc_source/environment-variables.md [hunk 5]
```diff
@@ -115,4 +115,37 @@ You can use the following environment variables that are accessible by default w
...
+
+1. Create a parameter name in the format `/amplify/{your_app_id}/{your_backend_environment_name}/{your_parameter_name}`
+
+1. Under type, choose Secure String for encrypted values
+
+1. Under KMS key source, choose my currenct account to use the default key for your account
+
+1. Under Value, enter the value you want
```
Comment (pananapread) [inline doc_source/environment-variables.md]: **Value**

Comment (pananapread) [reply doc_source/environment-variables.md]: Maybe it's just me, but I have no clue what value I'd add -- maybe an example?

Comment (letsbelopez) [reply doc_source/environment-variables.md]: a private key perhaps.

**Review by pananapread — APPROVED** [review]

File: doc_source/environment-variables.md [hunk 6]
```diff
@@ -154,6 +154,9 @@ Accessing an environment secret during a build is similar to environment variabl
 
 You can use the following environment secrets that are accessible by default within the Amplify Console\.
 
+Example full SSM parameter name:
```
Comment (pananapread) [inline doc_source/environment-variables.md]: The placement of this is confusing to me. Should it be under the table and say "Your full SSM parameter name should look something like this:"?

**Review by pananapread — CHANGES_REQUESTED** [review]

**Review by pananapread — APPROVED** [review]

File: doc_source/environment-variables.md (hunk 6 above)
Comment (pedramp20) [reply doc_source/environment-variables.md]: It is confusing to me too. Should the **your_parameter_name** in the SSM parameter format (/amplify/{your_app_id}/{your_backend_environment_name}/{your_parameter_name}) be same as the variable name in the table? If so please change the table column title to parameter name or mention it in the documentation.

**Pull request merged**